packages/remix-dev/vite/plugin.ts

```diff
@@ -977,21 +977,21 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
 
     return {
       version: String(Math.random()),
-      url: path.posix.join(
-        ctx.remixConfig.publicPath,
-        VirtualModule.url(browserManifestId)
-      ),
+      url: new URL(
+        VirtualModule.url(browserManifestId),
+        ctx.remixConfig.publicPath
+      ).href,
```

Comment:
I don't think `ctx.remixConfig.publicPath` is guaranteed to be an absolute URL here, is it?  If it's not, this will throw an exception:

```sh
> node
Welcome to Node.js v20.12.2.
Type ".help" for more information.
> new URL("file.txt", "/public")
Uncaught TypeError: Invalid URL
    at new URL (node:internal/url:796:36) {
  code: 'ERR_INVALID_URL',
  input: 'file.txt',
  base: '/public'
}
```